File: 🟡 Medium/1310 - XOR Queries of a Subarray/_1310XORQueriesofaSubarray.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int res[] = new int[queries.length];
        int i = 0;
        for (int[] query : queries) {
            int left = query[0];
            int right = query[1];
            int xorOfElem = arr[left];
            while (left < right) {
                left++;
                xorOfElem ^= arr[left];
            }
            res[i++] = xorOfElem;
        }
        return res;
    }
}

File: 🟡 Medium/1310 - XOR Queries of a Subarray/_1310XORQueriesofaSubarray.js
/**
 * @param {number[]} arr
 * @param {number[][]} queries
 * @return {number[]}
 */
var xorQueries = function (arr, queries) {
    let res = [];
    let i = 0;
    for (let query of queries) {
        let left = query[0];
        let right = query[1];
        let xorOfElem = arr[left];
        while (left < right) {
            left++;
            xorOfElem ^= arr[left];
        }
        res[i++] = xorOfElem;
    }
    return res;
};